scripts/SANS/SANSUtility.py

```diff
@@ -1025,6 +1025,60 @@ def create_add_algorithm(self, is_overlay):
             return PlusWorkspaces()
 
 
+def _clean_logs(ws, estimate_logs):
+    """
+    Remove bad proton charge times from the logs, if present.
+    :param ws: The workspace to clean
+    :param estimate_logs: bool. If true, estimate "good" values for bad proton logs
+    """
+    run = ws.getRun()
+    if run.hasProperty("proton_charge"):
+        pc = run.getProperty("proton_charge")
+        first_log_time = pc.firstTime().toISO8601String()
+        if int(first_log_time.split("-")[0]) < 1990:
```

Comment:
I feel we need to run this as a loop to handle the situation where there is more than one bum value.